### DISCUSSION TITLE
I have a custom CR10S

### DISCUSSION THREAD

**Discussion by jaredsay1974** [Jyers/Marlin#1924] [Q&A]
It has a ender3 V2 motherboard and screen and hotend all are ender3 v2 need help with firmwares on size and BL touch it I bought it used and never got it to work I'm losing my mind over it and I'm getting sick of  Throwing money after money into it I wish I could find someone in Minnesota to help me Troubleshoot it and get to work finely I have enough things to make another printer but that not my plains I only wanted two printer one small and one big I'm into radio control trucks cars boats I figure out how to setup a fix my ender3 v2 and that was a easy fix 😢 😞 😕 
![20210601_104123](https://user-images.githubusercontent.com/31258092/161195623-801c0dc7-1ab6-4533-9cf6-9be20930d830.jpg)
![IMG-20210523-WA0014](https://user-images.githubusercontent.com/31258092/161195690-7718ba95-d12a-4974-97e0-ea9c52085284.jpeg)

**Comment by BlueWings172** [comment]
@jaredsay1974 

You did not mention what is wrong with the printer. Why it is not working ? 

is the only issue that you installed an Ender 3 V2 board on it and don't have a firmware for it ? or you have issues with connecting the CR10s steppers and endstops and sensors to the Ender 3 v2 board ?

If the size of the bed is the only problem, you need change values in the firmware, compile it and then flash it.

I'm far from being an expert but this is what I think you need to do assuming that you have not compiled firmware before:

1- Check [this post](https://github.com/Jyers/Marlin/discussions/1787#discussioncomment-1978481) on instructions how to compile firmware. It is quite simple. You many want to start with the 3x3 High speed BlTouch in this folder: Marlin/Configuration Files/E3V2 Templates/BLTouch-3x3-HighSpeed/. It is recommended to change the 3x3 grid to 5x5. There is instructions in the first link how to do that.

2- Since the CR10s has a build volume of 300x300x400, you need to change these values in firmware. There is another great Youtube video by BV3D explaining which lines to be changed in the firmware. [This is the video,](https://www.youtube.com/watch?v=mjdp16ij3sU&t) the firmware parts starts at minute 9:00

3- Once the firmware is edited, compiled and flashed successfully, you need to input the probe offsets in the screen. Alternatively you can change it in the firmware code by changing the values in the #define NOZZLE_TO_PROBE_OFFSET { -44, -7, 0 } line. I noticed that your Bltouch probe is on the right side of the nozzle so probably your X value, will be positive. You can find many sources online how to determine these values.

4- Many things can go wrong, so you need to test that you axes are moving as they should and stopping when they should so you need to home the printer then from the screen move X, Y and Z to 300x300x400 positions. **Very important to keep your finger on the Power off switch so you can kill the power immediately when something goes wrong.**

5- If everything is working fine and you are confident that your axes are moving as they should, you can proceed with the rest of necessary steps that are done post firmware upgrade. This is my list, yours might be different:

1. Test X,Y, and Z axes movement and limits
2. PID Tuning for hotend and bed
3. Manual Bed Leveling with Probe
4. Z Offset
5. Calibrate E-Steps

Hope this helps.
